[TechDebt]: S3 bucket replication acceptance test panics

### Description

The `TestAccS3BucketReplicationConfiguration_schemaV2SameRegion` test panics during the destroy check:

```console
$ make testacc PKG=s3 TESTS=TestAccS3BucketReplicationConfiguration_schemaV2SameRegion
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/s3/... -v -count 1 -parallel 20 -run='TestAccS3BucketReplicationConfiguration_schemaV2SameRegion'  -timeout 180m
=== RUN   TestAccS3BucketReplicationConfiguration_schemaV2SameRegion
=== PAUSE TestAccS3BucketReplicationConfiguration_schemaV2SameRegion
=== CONT  TestAccS3BucketReplicationConfiguration_schemaV2SameRegion
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10126df90]

goroutine 2345 [running]:
github.com/aws/aws-sdk-go/service/s3.(*S3).newRequest(0x140000e8558?, 0x100d34774?, {0x10d8c9b80?, 0x14007ad2570?}, {0x10ca52d40?, 0x1400c718008?})
        /Users/jaredbaker/go/pkg/mod/github.com/aws/aws-sdk-go@v1.44.239/service/s3/service.go:100 +0x30
github.com/aws/aws-sdk-go/service/s3.(*S3).GetBucketReplicationRequest(0x140000e8528?, 0x14007ad2570)
        /Users/jaredbaker/go/pkg/mod/github.com/aws/aws-sdk-go@v1.44.239/service/s3/api.go:4142 +0xbc
github.com/aws/aws-sdk-go/service/s3.(*S3).GetBucketReplicationWithContext(0x140000e8748?, {0x10e1c36a0, 0x140033dd8c0}, 0x140000e86cc?, {0x0, 0x0, 0x1010140016681e0?})
        /Users/jaredbaker/go/pkg/mod/github.com/aws/aws-sdk-go@v1.44.239/service/s3/api.go:4196 +0x30
github.com/hashicorp/terraform-provider-aws/internal/service/s3_test.testAccCheckBucketReplicationConfigurationDestroyWithProvider.func1.1()
        /Users/jaredbaker/development/_worktrees/td-aws_s3-acctest-fixes/internal/service/s3/bucket_replication_configuration_test.go:1193 +0x38
github.com/hashicorp/terraform-provider-aws/internal/tfresource.RetryWhen.func1()
        /Users/jaredbaker/development/_worktrees/td-aws_s3-acctest-fixes/internal/tfresource/retry.go:30 +0x38
github.com/hashicorp/terraform-provider-aws/internal/tfresource.Retry.func1()
        /Users/jaredbaker/development/_worktrees/td-aws_s3-acctest-fixes/internal/tfresource/retry.go:215 +0x48
github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext.func1()
        /Users/jaredbaker/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/retry/state.go:110 +0x144
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext
        /Users/jaredbaker/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/retry/state.go:83 +0x1a4
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/s3 139.952s
FAIL
make: *** [testacc] Error 1
```

This is the only test exhibiting this behavior within the replication configuration test suite.

### References

_No response_

### Would you like to implement a fix?

None

--- a/internal/service/s3/bucket_replication_configuration_test.go
+++ b/internal/service/s3/bucket_replication_configuration_test.go
@@ -684,20 +684,17 @@ func TestAccS3BucketReplicationConfiguration_schemaV2(t *testing.T) {
 
 func TestAccS3BucketReplicationConfiguration_schemaV2SameRegion(t *testing.T) {
 	ctx := acctest.Context(t)
-	rName := sdkacctest.RandomWithPrefix("tf-acc-test")
-	rNameDestination := sdkacctest.RandomWithPrefix("tf-acc-test")
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	rNameDestination := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
 	dstBucketResourceName := "aws_s3_bucket.destination"
 	iamRoleResourceName := "aws_iam_role.test"
 	resourceName := "aws_s3_bucket_replication_configuration.test"
 
-	// record the initialized providers so that we can use them to check for the instances in each region
-	var providers []*schema.Provider
-
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:                 func() { acctest.PreCheck(ctx, t) },
 		ErrorCheck:               acctest.ErrorCheck(t, s3.EndpointsID),
-		ProtoV5ProviderFactories: acctest.ProtoV5FactoriesPlusProvidersAlternate(ctx, t, &providers),
-		CheckDestroy:             acctest.CheckWithProviders(testAccCheckBucketReplicationConfigurationDestroyWithProvider(ctx), &providers),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckBucketReplicationConfigurationDestroy(ctx),
 		Steps: []resource.TestStep{
 			{
 				Config: testAccBucketReplicationConfigurationConfig_schemaV2SameRegion(rName, rNameDestination),
@@ -1177,6 +1174,39 @@ func TestAccS3BucketReplicationConfiguration_migrate_withChange(t *testing.T) {
 			},
 		},
 	})
+}
+
+// testAccCheckBucketReplicationConfigurationDestroy is the equivalent of the "WithProvider"
+// version, but for use with "same region" tests requiring only one provider.
+func testAccCheckBucketReplicationConfigurationDestroy(ctx context.Context) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := acctest.Provider.Meta().(*conns.AWSClient).S3Conn()
+
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "aws_s3_bucket_replication_configuration" {
+				continue
+			}
+			input := &s3.GetBucketReplicationInput{Bucket: aws.String(rs.Primary.ID)}
+
+			output, err := tfresource.RetryWhenAWSErrCodeEquals(ctx, 2*time.Minute, func() (interface{}, error) {
+				return conn.GetBucketReplicationWithContext(ctx, input)
+			}, s3.ErrCodeNoSuchBucket)
+
+			if tfawserr.ErrCodeEquals(err, tfs3.ErrCodeReplicationConfigurationNotFound, s3.ErrCodeNoSuchBucket) {
+				continue
+			}
+
+			if err != nil {
+				return err
+			}
+
+			if replication, ok := output.(*s3.GetBucketReplicationOutput); ok && replication != nil && replication.ReplicationConfiguration != nil {
+				return fmt.Errorf("S3 Replication Configuration for bucket (%s) still exists", rs.Primary.ID)
+			}
+		}
+
+		return nil
+	}
 }
 
 func testAccCheckBucketReplicationConfigurationDestroyWithProvider(ctx context.Context) acctest.TestCheckWithProviderFunc {

